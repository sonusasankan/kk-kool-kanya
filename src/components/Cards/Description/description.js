import React from 'react';
import PropTypes from 'prop-types';

import Description from './style';
import getExcerpt from '../getExcerpt';


const description = ({
  readMore,
  excerptLength,
  children,
}) => (
  <Description className="card__description">
    {getExcerpt(children, excerptLength)}
    <strong className="readMore">Read More</strong>
  </Description>
);

description.propTypes = {
  children: PropTypes.string.isRequired,
  excerptLength: PropTypes.number,
  readMore: PropTypes.string.isRequired,
};

description.defaultProps = {
  excerptLength: 15
}

export default description;
