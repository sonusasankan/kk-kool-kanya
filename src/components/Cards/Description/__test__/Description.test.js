import React from 'react';

import { shallow } from 'enzyme';
import { default as CardDescription } from '../description';


describe('Description', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(
      <CardDescription readMore="https://netbramha.com/">
            this is a long string along with readmore link to limit number
      </CardDescription>,
    );
    expect(component).toMatchSnapshot();
  });
});
