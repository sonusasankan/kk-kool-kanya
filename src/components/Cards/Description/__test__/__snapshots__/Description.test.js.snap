// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <description
    readMore="https://netbramha.com/"
  >
    this is a long string along with readmore link to limit number
  </description>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        "this is a long string along with readmore link",
        <a
          href="https://netbramha.com/"
        >
           ...Read
        </a>,
      ],
      "className": "card__description",
    },
    "ref": null,
    "rendered": Array [
      "this is a long string along with readmore link",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " ...Read",
          "href": "https://netbramha.com/",
        },
        "ref": null,
        "rendered": " ...Read",
        "type": "a",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
    margin-top: 16px;
    font-size: 0.8rem;
",
        ],
      },
      "displayName": "styled.p",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "p",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          "this is a long string along with readmore link",
          <a
            href="https://netbramha.com/"
          >
             ...Read
          </a>,
        ],
        "className": "card__description",
      },
      "ref": null,
      "rendered": Array [
        "this is a long string along with readmore link",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " ...Read",
            "href": "https://netbramha.com/",
          },
          "ref": null,
          "rendered": " ...Read",
          "type": "a",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
    margin-top: 16px;
    font-size: 0.8rem;
",
          ],
        },
        "displayName": "styled.p",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "p",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
